CorsixTH/Lua/humanoid_actions/seek_room.lua

```diff
@@ -128,22 +128,26 @@ local function action_seek_room_no_treatment_room_found(room_type, humanoid)
   humanoid.waiting = 60
   local strings = _S.fax.disease_discovered_patient_choice
   -- Can this room be built right now? What is then missing?
-  -- TODO: Change to make use of Hospital:checkDiseaseRequirements
+
   local output_text = strings.can_not_cure
   -- TODO: can we really assert this? Or should we just make the patient go home?
-  local room = assert(humanoid.world.available_rooms[room_type], "room " .. room_type .. " not available")
+  --local room = assert(humanoid.world.available_rooms[room_type], "room " .. room_type .. " not available")
 
-  local room_discovered = false
-  if humanoid.hospital.discovered_rooms[room] then
-    room_discovered = true
-    local room_name, required_staff, staff_name = humanoid.world:getRoomNameAndRequiredStaffName(room_type)
-    if humanoid.hospital:hasStaffOfCategory(required_staff) then
-      output_text = strings.need_to_build:format(room_name)
+  local req = humanoid.hospital:checkDiseaseRequirements(humanoid.disease.id)
+
+  if req then
+    local room_name, _, staff_name = humanoid.world:getRoomNameAndRequiredStaffName(room_type)
+    if #req.rooms > 0 then
+      if next(req.staff) then
+        output_text = strings.need_to_build_and_employ:format(room_name, staff_name)
+      else
+        output_text = strings.need_to_build:format(room_name)
+      end
     else
-      output_text = strings.need_to_build_and_employ:format(room_name, staff_name)
+      output_text = strings.need_to_employ:format(staff_name)
     end
```

Comment:
Like Patient:updateMessage except we have a room type to get the staff string for this time.